hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -2840,7 +2848,8 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
      * if the write-behind queue has reached its per-node maximum
      * capacity.
      */
-    Map<K, Object> executeOnEntries(EntryProcessor entryProcessor, Predicate predicate);
+    <R> Map<K, R> executeOnEntries(@Nonnull EntryProcessor<? super K, ? super V, R> entryProcessor,
+                                   @Nonnull Predicate<K, V> predicate);
```

Comment:
`Predicate<? super K, ? super V>`

Reply:
`Predicate` and `Projection` improvements are reserved for a separate PR to keep this one focused and to minimise the number of changed lines as it's already pretty big.